uint8[] a
---
uint8[] a_out
